mlflow/tracking/_tracking_service/client.py

```diff
@@ -873,6 +873,25 @@ def download_artifacts(self, run_id, path, dst_path=None):
         """
         return self._get_artifact_repo(run_id).download_artifacts(path, dst_path)
 
+    def _log_url(self, run_id):
+        try:
+            host_url = self.store.get_host_creds().host
+        except AttributeError:
+            # If the store does not have host creds, we cannot log the URL. This is expected for
+            # SqlAlchemyStore.
+            return
+        run_info = self.store.get_run(run_id).info
```

Comment:
Is there a way to avoid making `get-run` request?

Reply:
missed this comment - with the current client design, we have to call `get_run()`. I don't worry much about it because `run_id` is DB index.

Reply:
> with the current client design, we have to call get_run()


Is this because we call this in `set_terminated`?

Reply:
similar to a comment above - putting logging in fluent API is not very robust. This feature request comes from Mosaic researchers, majority of whom are using mosaicml/composer, inside which it directly calls Mlflow client.

Reply:
Do they explicitly call `set_terminated`?

Reply:
yes they do, when they close the `trainer`